docs/guides/user-data.md

```diff
@@ -0,0 +1,111 @@
+```
+- `tf-cloud-init` - name given to the key;
+- `f `- according to this flag, `tf-cloud-init` and `tf-cloud-init.pub` files will be created in the current directory;
+- `email@example.com` - your email.
+
+**Windows Putty client**
+
+To generate an SSH key in Windows, use Putty and follow [these instructions](https://www.ssh.com/academy/ssh/putty/windows/puttygen).
+
+## Adding public SSH key to the script
+Open the `scripts/add-ssh-web-app.yaml` file in the cloned repository. Copy and paste the public key from the `tf-cloud-init.pub` into the `ssh_authorized_keys` parameter. Further, you will need to refer to this script in the server's `user_data` argument.
+```
+##...
+users:
+  - default
+  - name: terraform
+    gecos: terraform
+    primary_group: hashicorp
+    sudo: ALL=(ALL) NOPASSWD:ALL
+    groups: users, admin
+    ssh_import_id:
+    lock_passwd: false
+    ssh_authorized_keys:
+      -  # "Your public SSH key from the tf-cloud-init.pub"
+##...
+```
+For more details, you can read the [cloud-init documentation](https://cloudinit.readthedocs.io/en/latest/topics/format.html).
+
+## Adding the script to the user-data
+Go to the directory containing the `main.tf `file. In our case, it's the `serverscom-terraform`:
+```
+$ cd serverscom-terraform
+```
+Open the `main.tf` file and the `template_file` block and fill the `user_data` as it's described below:
+```
+data "template_file" "user_data" {
+  template = file("../scripts/add-ssh-web-app.yaml")
+ 
+resource "serverscom_dedicated_server" "node_1" {
+  hostname             = "node-1"
+  location             = "SJC1"
+  server_model         = "Dell R440 / 2xIntel Xeon Silver-4114 / 32 GB RAM / 1x480 GB SSD"
+  ram_size             = 32
+ 
+  operating_system     = "Ubuntu 20.04-server x86_64"
+ 
+  private_uplink       = "Private 10 Gbps with redundancy"
+  public_uplink        = "Public 10 Gbps with redundancy"
+  bandwidth            = "200000 GB"
+  # ...
+  # Some parameters are not displayed here to shorten the specification.
+  # You can see the complete example of the resource in the relevant section of the documentation.
+  # ...
+  user_data            = data.template_file.user_data.rendered
+   
+}
+```
+The `template_file` block refers to the `add-ssh-web-app.yaml` that will be processed as the `user-data` that will be initialized after provisioning.
+
+Save changes and make commands to initialize and apply the configuration:
+```
+$ terraform init && terraform apply
+```
+After a successful operation, you will have a dedicated server with a configured SSH key and Go web application.
+
+## Checking the result
+
+To connect to the server via SSH, use the following command in the terraform console:
+```
+$ ssh terraform@$(terraform output -raw public_ip) -i ../tf-cloud-init
+```
+- `public_ip` - public IPv4 address of the server.
+
+Thus, your SSH access is established without creating the keys on the provider's side.
+
+Move to the Go directory:
```

Comment:
According to previously specified user data, I can't find any reason why this directory should exist.